cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -648,12 +648,15 @@ def is_email_to_send_run_completion(self, status):
             if email_sending_job_submitted_interval is None:
                 # in case this didn't come with the token take the default value from the configuration
                 email_sending_job_submitted_interval = self.app.config.get('conf').email_sending_job_submitted_default_interval
-            if os.path.exists(self.scratch_dir + '/email_history'):
-                submitted_email_files = glob.glob(self.scratch_dir + '/email_history/email_*_submitted_*.email')
+
+            email_jobs_folders = glob.glob(f'scratch_sid_*_jid_{self.job_id}*/email_history')
+            if len(email_jobs_folders) >= 1:
+            # if os.path.exists(self.scratch_dir + '/email_history'):
+                submitted_email_files = glob.glob(f'scratch_sid_*_jid_{self.job_id}*/email_history/email_submitted_*.email')
                 if len(submitted_email_files) >= 1:
-                    last_submitted_email_sent = submitted_email_files[len(submitted_email_files) - 1]
-                    f_name, f_ext = os.path.splitext(os.path.basename(last_submitted_email_sent))
-                    time_last_email_submitted_sent = float(f_name.split('_')[3])
+                    times = []
+                    [times.append(float(os.path.splitext(os.path.basename(f))[0].split('_')[2])) for f in submitted_email_files]
```

Comment:
is this really a more undestandable version of two-line loop? :)

Reply:
if you do want to use list comprehesion, even if it is a long line, why not combine two lines and avoid append?

Reply:
I tried to compact a number of instructions. Sorry I didn't get what you mean with the combination of two lines instead of the append.

Reply:
I mean, if you do want to use list comprehension, is this not going to do what you want:

```
times = [ float(os.path.splitext(os.path.basename(f))[0].split('_')[2])) for f in submitted_email_files ]
```

while also using generated list.

Relying on side-effects of list comprehension is a bit unusual and arguably quite unpythonic. 
E.g. see some community discussion [here](https://stackoverflow.com/questions/5753597/is-it-pythonic-to-use-list-comprehensions-for-just-side-effects) and similar.

Maybe I am missing your intent.

Reply:
True, let's not going unpythonic. Didn't consider these, actually very important, aspect of the side-effects that could actually be dangerous.